From d7654b82b2796580b11bd5ba83c3048b1053f648 Mon Sep 17 00:00:00 2001
From: Davide Garberi <dade.garberi@gmail.com>
Date: Wed, 24 Apr 2019 19:45:58 +0200
Subject: [PATCH 4/4] releasetools: Correctly assert for A-only system-as-root

* As of aosp A-only system-as-root in recovery the system partition should be mounted to /system_root and /system should be a symlink of /system_root/system
* Add a check for the presence of either / or /system_root in the recovery fstab to not break any system-as-root device

Change-Id: Ia9ee66c381f91ca262e6374dd365157aefe435ed
Signed-off-by: Davide Garberi <dade.garberi@gmail.com>
---
 tools/releasetools/common.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index df077b18e..62efb9568 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -347,8 +347,12 @@ def LoadRecoveryFSTab(read_helper, fstab_version, recovery_fstab_path,
   # system. Other areas assume system is always at "/system" so point /system
   # at /.
   if system_root_image:
-    assert not d.has_key("/system") and d.has_key("/")
-    d["/system"] = d["/"]
+    if d.has_key("/"):
+      assert not d.has_key("/system") and d.has_key("/")
+      d["/system"] = d["/"]
+    elif d.has_key("/system_root"):
+      assert not d.has_key("/system") and d.has_key("/system_root")
+      d["/system"] = d["/system_root"]
   return d
 
 
-- 
2.17.1

